lca-cli/postpivot/postpivot.go

```diff
@@ -541,3 +554,47 @@ func (p *PostPivot) applyNMStateConfiguration(seedReconfiguration *clusterconfig
 
 	return nil
 }
+
+// createPullSecretFile creates auth file on filesystem in order to be able to pull images
+// and runs createPullSecretManifest to write secret in manifests folder
+func (p *PostPivot) createPullSecretFile(pullSecret, pullSecretFile, pullSecretManifest string) error {
+	// TODO: Should return error in the future as cluster will not be operational without it
+	if pullSecret == "" {
+		p.log.Infof("Pull secret was not provided")
+		return nil
+	}
+	p.log.Infof("Move seed PS file aside")
+	if err := utils.MoveFileIfExists(pullSecretFile, pullSecretFile+"_seed"); err != nil {
+		return err
+	}
+
+	p.log.Infof("Writing provided pull secret to %s", pullSecretFile)
+	if err := os.WriteFile(pullSecretFile, []byte(pullSecret), 0o600); err != nil {
+		return fmt.Errorf("failed to write pull secret to %s, err %w", pullSecret, err)
+	}
+
+	return p.createPullSecretManifest(pullSecret, pullSecretManifest)
```

Comment:
Consider moving this function call to wherever you call `createPullSecretFile`.
Or rename `createPullSecretFile` to `createPullSecretFileAndManifest`